blockchain/fullblocktests/generate.go

```diff
@@ -2149,6 +2160,43 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
 	})
 	rejected(blockchain.ErrImmatureSpend)
 
+	// Create block with a malformed outputs order for a ticket purchase.
+	//
+	//   ... -> brs3(14)
+	//                  \-> bmf36(15)
+	g.SetTip("brs3")
+	g.NextBlock("bmf36", outs[15], ticketOuts[15], func(b *wire.MsgBlock) {
+		prevBlock := g.Tip()
+		spendOut := chaingen.MakeSpendableOut(prevBlock, 1, 0)
+		ticketPrice := dcrutil.Amount(g.CalcNextRequiredStakeDifficulty())
+		ticket := g.CreateTicketPurchaseTx(&spendOut, ticketPrice, lowFee)
+		tmp := ticket.TxOut[0]
```

Comment:
No need to do this in Go since it supports multiple assignment.  e.g.

```Go
ticket.TxOut[0], ticket.TxOut[2] = ticket.TxOut[2], ticket.TxOut[0]
```